core/domain/user_jobs_one_off.py

```diff
@@ -81,6 +81,64 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class PopulateUserAuthModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating and populating UserAuthModel for
+    all registered users.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        """Marks a job as queued and adds it to a queue for processing."""
+        super(PopulateUserAuthModelOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+
+        # Assumptions:
+        # 1) We are only having full users in the database, and no
+        #    profile users. For profiles we need to do a role check and ensure
+        #    the gae_id attribute is None for them.
+        # 2) This migration works correctly only if gae_id cannot be updated
+        #   in any way for a user. If a gae_id is mapped to some other user by
+        #   deleing it, an error will be raised while migration (see below).
+
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, item.gae_id)
+        if user_auth_model is not None:
+            if user_auth_model.id != item.id:
+                yield (
+                    'Error, gae_id already existing',
+                    {'user_id': item.id, 'gae_id': item.gae_id}
+                )
+            else:
+                if (item.deleted is True and
+                        user_auth_model.deleted is not True):
+                    user_auth_model.deleted = item.deleted
+                    user_auth_model.put()
+                    yield (
+                        'Updated', {'user_id': item.id, 'gae_id': item.gae_id})
+        else:
+            user_models.UserAuthModel(
+                id=item.id,
+                gae_id=item.gae_id,
+                deleted=item.deleted
+            ).put()
+            yield ('Created', {'user_id': item.id, 'gae_id': item.gae_id})
```

Comment:
'SUCCESS - Created UserAuth model'

The second arg should be a string (see other jobs for examples).

Reply:
I think it can also be a tuple or some other object, I think I used tuples in the user ID migration.

Reply:
Oh, sorry -- thanks for catching that @vojtechjelinek, I was thinking of reduce(). Yes this is fine but if you do that, convert it to a string in reduce(). (Having said that it is probably easier to just use a string here, given that your reduce() is trivial.)

Reply:
Returning just `1` now as we just want to number of users populated as per the other comment.